src/PresentationalComponents/Modals/AddEditTopic.js

```diff
@@ -0,0 +1,193 @@
+import './_AddEditTopic.scss';
+
+import React, { useState, useEffect, useRef } from 'react';
+import { injectIntl } from 'react-intl';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import {
+    Modal, Form, FormGroup, Button, Split,
+    SplitItem, TextInput, TextArea, Radio, Checkbox
+} from '@patternfly/react-core';
+
+import messages from '../../Messages';
+import { setTopic, deleteTopic } from '../../AppActions';
+
+const AddEditTopic = ({ handleModalToggle, intl, isModalOpen, topic, setTopic, deleteTopic }) => {
+
+    const [name, setName] = useState();
+    const [description, setDescription] = useState();
+    const [tag, setTag] = useState();
+    const [enabled, setEnabled] = useState(false);
+    const [featured, setFeatured] = useState();
+
+    const setDeleteHandler = ({ type, options }) => {
+        type === 'DELETE' ?
+            deleteTopic(options) :
+            setTopic(options);
+        handleModalToggle(false);
+    };
+
+    const footer = (
+        <Split gutter='sm' style={{ width: '100%' }}>
+            <SplitItem>
+                <Button key='confirm' variant='primary' onClick={ () => setDeleteHandler({
```

Comment:
yah do this twice, `setDeleteHandler` maybe abstract as a function somewhere else and pass in the type?